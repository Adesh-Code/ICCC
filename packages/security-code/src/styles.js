const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
