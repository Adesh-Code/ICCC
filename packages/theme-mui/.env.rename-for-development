# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE