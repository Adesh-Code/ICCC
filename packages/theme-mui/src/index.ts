// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
