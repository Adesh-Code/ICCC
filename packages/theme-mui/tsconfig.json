{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
