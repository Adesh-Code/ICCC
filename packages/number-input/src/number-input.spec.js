import React from "react";
import PropTypes from "prop-types";
import { screen, render, fireEvent } from "@testing-library/react";
import { createSerializer, matchers } from "@emotion/jest";
import NumberInput from "./number-input";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

expect.extend(matchers);
expect.addSnapshotSerializer(createSerializer({ includeStyles: false }));

const arrange = (args) => {
  return render(
    <ThemeProvider theme={light}>
      <NumberInput {...args} />
    </ThemeProvider>
  );
};

const NumberInputMock = ({ value, ...otherProps }) => {
  const [num, setNum] = React.useState(value);
  return (
    <ThemeProvider theme={light}>
      <NumberInput value={num} onChange={setNum} {...otherProps} />
    </ThemeProvider>
  );
};
NumberInputMock.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

describe("NumberInput component", () => {
  it("renders a number input", () => {
    const { container } = arrange();
    expect(container).toMatchSnapshot();
  });
  it("accepts classnames", () => {
    const { container } = arrange({ className: "foo" });
    expect(container.firstChild).toHaveClass("foo");
    expect(container).toMatchSnapshot();
  });
  it("accepts error prop", () => {
    const { container } = arrange({ error: true });
    expect(container.firstChild).toHaveClass("Mui-error");
    expect(container).toMatchSnapshot();
  });
  it("accepts a controled value", () => {
    const { container } = arrange({ value: 32 });
    expect(screen.getByDisplayValue("32")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("changes the inputs value if a number is written", () => {
    const { container } = render(<NumberInputMock />);
    const input = screen.getByDisplayValue("0");
    fireEvent.change(input, { target: { value: 27 } });
    expect(screen.getByDisplayValue("27")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("accepts negative values", () => {
    const { container } = render(<NumberInputMock />);
    const input = screen.getByDisplayValue("0");
    fireEvent.change(input, { target: { value: -27 } });
    expect(screen.getByDisplayValue("-27")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("returns an empty string if 'NaN' is passed as the input value", () => {
    const { container } = render(<NumberInputMock />);
    const input = screen.getByDisplayValue("0");
    fireEvent.change(input, { target: { value: Number("sdfgdf") } });
    expect(screen.getByDisplayValue("")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("empty strings", () => {
    const { container } = render(<NumberInputMock />);
    const input = screen.getByDisplayValue("0");
    fireEvent.change(input, { target: { value: "" } });
    expect(screen.getByDisplayValue("")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("non numerical characters", () => {
    const { container } = render(<NumberInputMock />);
    const input = screen.getByDisplayValue("0");
    fireEvent.change(input, { target: { value: "home" } });
    expect(screen.getByDisplayValue("")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("decrease the amount by click the minus button ", () => {
    const { container } = render(<NumberInputMock value={2} />);
    const subtractButton = container.querySelectorAll(".MuiSvgIcon-root")[0];
    fireEvent.click(subtractButton);
    expect(screen.getByDisplayValue("1")).toBeInTheDocument();
  });
  it("increase the amount by click the plus button", () => {
    const { container } = render(<NumberInputMock value={2} />);
    const addButton = container.querySelectorAll(".MuiSvgIcon-root")[1];
    fireEvent.click(addButton);
    fireEvent.click(addButton);
    expect(screen.getByDisplayValue("4")).toBeInTheDocument();
  });
  it("adjusts to the maximum value when clicking a button if the current value exceeds the maximum value", () => {
    const { container } = render(<NumberInputMock max={10} value={12} />);
    const addButton = container.querySelectorAll(".MuiSvgIcon-root")[1];
    fireEvent.click(addButton);
    expect(screen.getByDisplayValue("10")).toBeInTheDocument();
  });
  it("adjusts to the minimun value when clicking a button if the current value is less the minimun value", () => {
    const { container } = render(<NumberInputMock min={10} value={8} />);
    const addButton = container.querySelectorAll(".MuiSvgIcon-root")[1];
    fireEvent.click(addButton);
    expect(screen.getByDisplayValue("10")).toBeInTheDocument();
  });
  it("accepts floating point numbers", () => {
    const { container } = render(<NumberInputMock step={1.3} />);
    const input = screen.getByDisplayValue("0");
    fireEvent.change(input, { target: { value: "3.2" } });
    expect(screen.getByDisplayValue("3.2")).toBeInTheDocument();
    const subtractButton = container.querySelectorAll(".MuiSvgIcon-root")[0];
    fireEvent.click(subtractButton);
    expect(screen.getByDisplayValue("1.9")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("accepts a custom onKeyDown function", () => {
    const onKeyDown = jest.fn();
    render(<NumberInputMock onKeyDown={onKeyDown} />);
    const input = screen.getByDisplayValue("0");
    fireEvent.keyDown(input, { key: "Enter", code: 13 });
    expect(onKeyDown).toHaveBeenCalled();
  });
  it("should not update when another key is pressed", () => {
    const onChangeMock = jest.fn();
    render(<NumberInputMock />);
    const input = screen.getByDisplayValue("0");
    fireEvent.keyDown(input, { key: "Enter", code: 13 });
    expect(onChangeMock).not.toHaveBeenCalled();
  });
  it("does not trigger the onChange function when its not passed", () => {
    arrange();
    const input = screen.getByDisplayValue("0");
    fireEvent.change(input, { target: { value: 3 } });
    expect(screen.getByDisplayValue("3")).toBeInTheDocument();
  });
  it("will change to the minimum value if an empty string value is entered and the buttons are pressed", () => {
    const { container } = render(<NumberInputMock min={10} />);
    const input = screen.getByDisplayValue("0");
    fireEvent.change(input, { target: { value: "" } });
    expect(screen.getByDisplayValue("")).toBeInTheDocument();
    const addButton = container.querySelectorAll(".MuiSvgIcon-root")[1];
    fireEvent.click(addButton);
    expect(screen.getByDisplayValue("10")).toBeInTheDocument();
  });
});
