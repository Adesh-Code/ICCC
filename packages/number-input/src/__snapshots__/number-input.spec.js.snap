// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput component accepts a controled value 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root DhigNumberInput  MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd emotion-0"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
      id=""
      type="number"
      value="32"
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/subtract"
          >
            <path
              d="M4 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/add"
          >
            <path
              d="M12 4V12M12 20V12M12 12H4M12 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
  </div>
</div>
`;

exports[`NumberInput component accepts classnames 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root DhigNumberInput foo MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd emotion-0"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
      id=""
      type="number"
      value="0"
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/subtract"
          >
            <path
              d="M4 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/add"
          >
            <path
              d="M12 4V12M12 20V12M12 12H4M12 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
  </div>
</div>
`;

exports[`NumberInput component accepts error prop 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root DhigNumberInput  MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-adornedEnd emotion-0"
  >
    <input
      aria-invalid="true"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
      id=""
      type="number"
      value="0"
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/subtract"
          >
            <path
              d="M4 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/add"
          >
            <path
              d="M12 4V12M12 20V12M12 12H4M12 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
  </div>
</div>
`;

exports[`NumberInput component accepts floating point numbers 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root DhigNumberInput  MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd emotion-0"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
      id=""
      type="number"
      value="1.9"
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/subtract"
          >
            <path
              d="M4 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/add"
          >
            <path
              d="M12 4V12M12 20V12M12 12H4M12 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
  </div>
</div>
`;

exports[`NumberInput component accepts negative values 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root DhigNumberInput  MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-adornedEnd emotion-0"
  >
    <input
      aria-invalid="true"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
      id=""
      type="number"
      value="-27"
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/subtract"
          >
            <path
              d="M4 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/add"
          >
            <path
              d="M12 4V12M12 20V12M12 12H4M12 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
  </div>
</div>
`;

exports[`NumberInput component changes the inputs value if a number is written 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root DhigNumberInput  MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd emotion-0"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
      id=""
      type="number"
      value="27"
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/subtract"
          >
            <path
              d="M4 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/add"
          >
            <path
              d="M12 4V12M12 20V12M12 12H4M12 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
  </div>
</div>
`;

exports[`NumberInput component empty strings 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root DhigNumberInput  MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-adornedEnd emotion-0"
  >
    <input
      aria-invalid="true"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
      id=""
      type="number"
      value=""
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/subtract"
          >
            <path
              d="M4 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/add"
          >
            <path
              d="M12 4V12M12 20V12M12 12H4M12 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
  </div>
</div>
`;

exports[`NumberInput component non numerical characters 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root DhigNumberInput  MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-adornedEnd emotion-0"
  >
    <input
      aria-invalid="true"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
      id=""
      type="number"
      value=""
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/subtract"
          >
            <path
              d="M4 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/add"
          >
            <path
              d="M12 4V12M12 20V12M12 12H4M12 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
  </div>
</div>
`;

exports[`NumberInput component renders a number input 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root DhigNumberInput  MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd emotion-0"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
      id=""
      type="number"
      value="0"
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/subtract"
          >
            <path
              d="M4 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/add"
          >
            <path
              d="M12 4V12M12 20V12M12 12H4M12 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
  </div>
</div>
`;

exports[`NumberInput component returns an empty string if 'NaN' is passed as the input value 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root DhigNumberInput  MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-adornedEnd emotion-0"
  >
    <input
      aria-invalid="true"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
      id=""
      type="number"
      value=""
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/subtract"
          >
            <path
              d="M4 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd emotion-2"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="utility/add"
          >
            <path
              d="M12 4V12M12 20V12M12 12H4M12 12H20"
              id="stroke"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </g>
        </svg>
      </svg>
    </div>
  </div>
</div>
`;
